src/apps/code-editor/src/app/components/SideBar/index.tsx

```diff
@@ -19,8 +23,17 @@ type NavType = "view" | "script" | "stylesheet" | "file";
 interface SideBarProps {
   navCode: NavCode;
   dispatch: (action: any) => void;
+  openCreateFileDialog?: (type: string, nav: NavType) => void;
```

Comment:
Why is this optional? I assume this component is being used at only one location and we'd most likely want to be able to do something when a user clicks on the open file dialog button. I also see that we're not checking if this exists before the execution below which would error out.